### PR TITLE
Fix #1926

### LINKED ISSUE #1926: PlayerJoinEvent, Player::setAllowFlight
#### Issue description

I'm tried to call Player::setAllowFlight(true) in PlayerJoinEvent, and it's not works.

#### Steps to reproduce the issue

1. Create event `PlayerJoinEvent` in plugin.
2. In this event, call `$event->getPlayer()->setAllowFlight(true);`.
3. Try to fly.

You can see problem here `Player.php:1008`:
```
		$this->server->getPluginManager()->callEvent($ev = new PlayerJoinEvent($this, new TranslationContainer(TextFormat::YELLOW . "%multiplayer.player.joined", [
			$this->getDisplayName()
		])));

		if(strlen(trim($msg = $ev->getJoinMessage())) > 0){
			if($this->server->playerMsgType === Server:: PLAYER_MSG_TYPE_MESSAGE) $this->server->broadcastMessage($msg);
			elseif($this->server->playerMsgType === Server::PLAYER_MSG_TYPE_TIP) $this->server->broadcastTip(str_replace("@player", $this->getName(), $this->server->playerLoginMsg));
			elseif($this->server->playerMsgType === Server::PLAYER_MSG_TYPE_POPUP) $this->server->broadcastPopup(str_replace("@player", $this->getName(), $this->server->playerLoginMsg));
		}

		$this->setAllowFlight($this->gamemode == 3 || $this->gamemode == 1);
```

#### OS and versions

* Genisys: https://github.com/iTXTech/Genisys/commit/eb168db2e3a32097ba3d654bbcafc3e7d113efa8
* PHP: 7.0.1
* Server OS: Windows 10, Ubuntu Linux
* Game version: 0.15.6

#### Crashdump, backtrace, memory dumps, plugins list or other files

```
```

### PR DESCRIPTION
**Pull request by Tarik02**
### Description

This fixes bugs, when plugins calls `Player::setAllowFlight`, `Player::setHealth` or `Player::setFood` in `PlayerJoinEvent`.

### Reason to modify

I'm want to use this methods in `PlayerJoinEvent`.

### Tests & Reviews

I have tested the code and it works.

### CODE REVIEW

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -998,6 +998,12 @@ protected function doFirstSpawn(){
 		}
 
 		$this->teleport($pos);
+		
+		$this->setAllowFlight($this->gamemode == 3 || $this->gamemode == 1);
+		
+		$this->setHealth($this->getHealth());
+		if($this->server->foodEnabled) $this->setFood($this->getFood());
+		else $this->setFood(20);
```
Comment (dktapps) [inline src/pocketmine/Player.php]: This condition is redundant. Please remove it.

File: src/pocketmine/Player.php [hunk 2]
```diff
@@ -998,7 +998,12 @@ protected function doFirstSpawn(){
 		}
 
 		$this->teleport($pos);
-
+		
+		$this->setAllowFlight($this->gamemode == 3 || $this->gamemode == 1);
```
Comment (SOF3) [inline src/pocketmine/Player.php]: This is actually `isCreative()`.

Comment (dktapps) [inline src/pocketmine/Player.php]: I don't like that method name, it's misleading. I tend to use getGamemode() & 0x01

**Comment by ishitatsuyuki** [comment]
Please resolve the conflicts, and squash your changes.

**Pull request closed without merging**